js/everything.js

```diff
@@ -225,7 +230,44 @@ var ZenPlayer = {
                 updateTweetMessage();
             });
 
-            plyrPlayer.addEventListener("timeupdate", function() {
+            plyrPlayer.addEventListener("ended", function () {
+                videoMetadata = getParsedvideoMetadata();
+                var index = -1;
+                if (videoMetadata) {
+                    var hasDuplicates = function (arr) {
+                        var len = arr.length;
+                        for (var i = 0; i < len; i++) {
+                            if (arr[i].id === currentVideoID) {
+                                return i;
+                            }
+                        }
+                        return -1;
+                    };
+                    index = hasDuplicates(videoMetadata.items);
+
+                    if (index !== -1) {
+                        videoMetadata["items"].splice(index, 1);
```

Comment:
If we splice, we're actually losing track of videos we've already played.
What if we instead store 2 arrays: played videos, and suggested videos.
When we pick the next suggested video, ensure it's not in the played videos list - if it is remove it.
When we query the API for suggested videos, only add the videos if they're not already in either list.

Reply:
here I am just changing the order of playlist since if duplicate exist then I have to change the order since this playlist was requested again so I put it on the top, so there is no need of one new playlist